lib/wallet/wallet.js

```diff
@@ -1820,14 +1824,77 @@ class Wallet extends EventEmitter {
     }
   }
 
+  /**
+   * Create and finalize a bid & a reveal (in advance)
+   * MTX with a lock.
+   * @param {String} name
+   * @param {Number} value
+   * @param {Number} lockup
+   * @param {Object} options
+   */
+
+  async createAuctionTxs(name, value, lockup, options) {
+    const unlock = await this.fundLock.lock();
+    try {
+      return await this._createAuctionTxs(name, value, lockup, options);
+    } finally {
+      unlock();
+    }
+  }
+
+  /**
+   * Create and finalize a bid & a reveal (in advance)
+   * MTX without a lock.
+   * @param {String} name
+   * @param {Number} value
+   * @param {Number} lockup
+   * @param {Object} options
+   */
+
+  async _createAuctionTxs(name, value, lockup, options) {
+    const bid = await this._createBid(name, value, lockup, options);
+
+    // Prepare the data needed to make the reveal in advance
+    const rawName = Buffer.from(name, 'ascii');
+    const nameHash = rules.hashName(rawName);
+    const bidBlind = new BlindBid();
+    bidBlind.name = name;
+    bidBlind.nameHash = nameHash;
+    bidBlind.prevout = Outpoint.fromTX(bid, 0);
+    bidBlind.value = bid;
+    bidBlind.lockup = lockup;
+    bidBlind.blind = bid.outputs[0].covenant.getHash(3);
+    bidBlind.own = true;
+    const coins = [];
+    const bidCoin = Coin.fromTX(bid, 0, this.wdb.height);
+    coins.push(bidCoin);
+
+    const acct = options ? options.account : null;
+    const force = options ? options.force || false : false;
```

Comment:
I think this can just be `const force = options.force` ?

But actually if we go with my suggestion of copying the reveal logic to this new function and leaving `makeReveal()` alone as-is, we won't need the force option at all. (This function will just always create a pre-signed reveal without doing any context checks).